### DISCUSSION TITLE
Additional fonts support in rendering PDF as ppm

### DISCUSSION THREAD

**Discussion by krishnanunnik** [pymupdf/PyMuPDF#1319] [Looking for help]
I am using the PyMuPDF(1.18.19) for converting the PDF pages to images. My code for the same is below.

    try:
        pdf_doc = fitz.Document(None, pdf_file, "pdf")
    except Exception:
        raise FormatConversionError('PDF not valid')
    
    page_indices = list(range(0, pdf_doc.pageCount))
    images = []
    
    for page_no in page_indices:
        page = pdf_doc.loadPage(page_no)

        width, height = page.MediaBox.height, page.MediaBox.width
        zoom_ratio = 1000 / height
        page_img = page.getPixmap(matrix=fitz.Matrix(zoom_ratio, zoom_ratio)).getImageData("ppm")
        images.append(page_img)

This code works fine for most of the PDF documents. But for certain PDFs which contains fonts like Noto Sans, Calibri and few other fonts, the image is not generated properly. Looks like below
[![Invalid Image][2]][2]

I understand this problem as a issue with Fonts support. There was a reference to install the `pymupdf-fonts` package to solve this. But that also did not work for this case. 

Is there any way, I can add the font support to the library using the ttf files  ? Any clues will be helpful!

Other information:

 - python 3.7.9
 - Ubuntu 18 64-bit

Thanks in Advance

  [1]: https://github.com/pymupdf/PyMuPDF
  [2]: https://i.stack.imgur.com/5sdO3.png

**Comment by JorjMcKie** [comment]
Sounds a it weird - please let me have a small example PDF.

**Reply by krishnanunnik** [reply]
I'm sorry @JorjMcKie , I cannot share the document due to privacy reasons. But I can share you some observations. 
I created a PDF using the same fonts in OpenOffice , and that worked fine with PyMuPDF. So I digged a little more and understood, that the fonts are encoded in Identity-H, for the document with the issue.

I dont know much about the encoding and its effect. But thought it will be good to mention. 

Details on problematic PDF. 

![image](https://user-images.githubusercontent.com/1677896/137101519-124fbb50-e6a6-4cc0-b133-ed8589c473a2.png)



A custom created PDF with same fonts

![image](https://user-images.githubusercontent.com/1677896/137101646-6e398ae5-08da-4e99-ae16-3f1eb49298b9.png)

Does this give any more info to you ?

**Reply by JorjMcKie** [reply]
> I cannot share the document due to privacy reasons.

Too bad.

> Does this give any more info to you ?

Not really.
The one thing you can try is whether the coming version 1.19.0 does a better job. I am going to prepare a wheel and let you know from where to download.

**Reply by JorjMcKie** [reply]
Please find your wheel [here](https://github.com/JorjMcKie/py-mupdf/actions/runs/1336973495). At the page bottom there is an "Artifact", which is the wheel wrapped in an extra ZIP.

**Reply by krishnanunnik** [reply]
Thanks. I will check and let you know the result.
